rocksdb/rocksdb-overview.md

```diff
@@ -0,0 +1,61 @@
+---
+title: RocksDB Overview
+summary: Learn the basic concepts and working principles of RocksDB.
+category: reference
+---
+
+# RocksDB Overview
+
+[RocksDB](https://github.com/facebook/rocksdb) is an LSM-tree storage engine that provides key-value store and read-write functions. It is developed by Facebook and based on LevelDB. Key-value pairs written by the user are firstly inserted into Write Ahead Log (WAL) and then written to the SkipList in memory (a data structure called MemTable). LSM-tree engines convert the random modification (insertion) to sequential writes to the WAL file, so they provide better write throughput than B-tree engines.
+
+Once the data in memory reaches a certain size, RocksDB flushes the content into a Sorted String Table (SST) file in the disk. SST files are organized in multiple levels (the default is up to 6 levels). When the total size of a level reaches the threshold, RocksDB chooses part of the SST files and merges them into the next level. Each subsequent level is 10 times larger than the previous one, so 90% of the data is stored in the last layer.
+
+RocksDB allows users to create multiple Column Families. Column Families have their own SkipList and SST files, and they share the same WAL file. In this way, different Column Families can have different settings according to the application characteristics. It does not increase the number of writes to WAL at the same time.
+
+## TiKV architecture
+
+The architecture of TiKV is illustrated as follows:
+
+![TiKV RocksDB](/media/tikv-rocksdb.png)
+
+As the storage engine of TiKV, RocksDB is used to store Raft logs and user data. All data in a TiKV node shares two RocksDB instances. One is for Raft log (often called raftdb), and the other is for user data and MVCC metadata (often called kvdb). There are four Column Families (CF) in kvdb: raft, lock, default, and write:
```

Comment:
`Column Families (CF)` or `Column Families (CFs)`, or ???

Reply:
`Column Families (CFs)` is better. Good catch.